pydantic/dataclasses.py

```diff
@@ -37,13 +37,13 @@ def dataclass(
         config: ConfigDict | type[object] | None = None,
         validate_on_init: bool | None = None,
         kw_only: bool = ...,
-    ) -> Callable[[type[_T]], type[PydanticDataclass]]:
```

Comment:
This `Any` and the one below are the sources of the `Expression type contains "Any" (has type overloaded function)` mypy errors, and I suspect are why it weirdly adds the _T instead of Any.

Seems like a bit of an abuse of mypy, but considering it's a user-facing error, not just an us-facing error, I think it's worth just doing a # type: ignore for pyright